trackpy/tests/test_linking.py

```diff
@@ -653,9 +697,50 @@ def test_drop_link(self):
 
 class TestNumbaLink(SubnetNeededTests):
     def setUp(self):
+        _skip_if_no_numba()
         self.linker_opts = dict(link_strategy='numba')
 
 
 class TestNonrecursiveLink(SubnetNeededTests):
     def setUp(self):
         self.linker_opts = dict(link_strategy='nonrecursive')
+
+
+class TestBTreeLink(SubnetNeededTests):
+    def setUp(self):
+        _skip_if_no_sklearn()
+        self.linker_opts = dict(neighbor_strategy='BTree')
+
+    def test_custom_dist_func(self):
+        # Several 2D random walkers
+        N = 5
+        length = 5
+        step_size = 2
+        search_range = 3
+
+        steps = (np.random.random((2, length, N)) - 0.5) * step_size
+        x, y = np.cumsum(steps, axis=2)
+        f = DataFrame(dict(x=x.ravel(), y=y.ravel(),
+                           frame=np.repeat(np.arange(length), N)))
+
+        # link in normal (2D Euclidean) coordinates
+        expected = self.link(f, search_range)
+
+        # compute radial coordinates
+        f_radial = f.copy()
+        f_radial['angle'] = np.arctan2(f_radial['y'], f_radial['x'])
+        f_radial['r'] = np.sqrt(f_radial['y']**2 + f_radial['x']**2)
+        # leave x, y for the comparison at the end
+
+        def dist_func(a, b):
```

Comment:
Upon merge, we should create an issue to write a tutorial or a short example about how to do this.